expat/lib/expat.h

```diff
@@ -124,9 +124,64 @@ enum XML_Error {
   XML_ERROR_RESERVED_PREFIX_XMLNS,
   XML_ERROR_RESERVED_NAMESPACE_URI,
   /* Added in 2.2.1. */
-  XML_ERROR_INVALID_ARGUMENT
+  XML_ERROR_INVALID_ARGUMENT,
+  /* Added in 2.3.0 */
+  XML_ERROR_ENTITY_VIOLATION_SIZE,
+  XML_ERROR_ENTITY_VIOLATION_NESTED_SIZE,
+  XML_ERROR_ENTITY_VIOLATION_RATIO,
+  XML_ERROR_ENTITY_VIOLATION_DEPTH,
+  XML_ERROR_HASH_TABLE_VIOLATION
```

Comment:
Maybe `XML_ERROR_HASH_TABLE_VIOLATION_SIZE` with `_SIZE` appended so that it's clear what the violation is about?

Reply:
``XML_ERROR_HASH_TABLE_SIZE_VIOLATION`` ?

Reply:
Would sound good on its own but not match the `*_VIOLATION_*` schema above. As you prefer.